### DISCUSSION TITLE
Additional PCB Headers that are needed for Sensors

### DISCUSSION THREAD

**Discussion by Tysonpower** [Gifford47/HCPBridgeMqtt#27] [Ideas]
I got a message from @14yannick requesting that i add some pcb headers for additional Sensors to my next PCB Revision.

I would like to discuss here what exactly is needed in terms of Sensors and COnnection points so i can add them to my next PCB Revison.

If possible i prefer general IO ports instead of some special pinout, but if a sensor just needs some general pin header like GND, 3V3 and two IO Pins i'm happy to add them.

All sensors need to be 3V3, so no 5V avaliable on my boards.

**Comment by 14yannick** [comment]
Hi, I saw your pcb in the HA Community [forum](https://community.home-assistant.io/t/hormann-garage-door-via-mqtt/279938/279).
Nice work, You haven't choose the easy way:-) using the chip instead the Dev board. Is your target also to move the component of the RS485 to the PCB? Does it fit in the Motor compartment? for the 5v I need to check but I think everything should be possible in 3.3v.
I will make some drawing how I imagine the pin outs for the sensors.

**Comment by Tysonpower** [comment]
I Will move back to the RS Module, it's cheaper to buy them and more easy for the PCB.
The next Revision will also have a different PSU circuit that is smaller and lighter but still provides up to 3A at 3V3.

The boards are currently 55x55mm, with the new PSU i can make them even smaller if we don't add to many Sensors.

Is the plan t solder the sensor directly to the pin headers or with a cable? Need to know if should leave some space for them on the pcb or if i only need the headers.

**Comment by Gifford47** [comment]
In a few weeks, my sensor boards are also coming home:) they have all current implemented sensors on board (sensors are picked up as modules).
Damn, I didn't know anything about your new PCB :)

**Reply by Tysonpower** [reply]
just a sensor pcb or also based on the psu moduel and esp32 devkit?

**Reply by Gifford47** [reply]
It has an own lm2596 PSU with an esp32 and chip. The rs485 was added as the known module. But I also like your current PSB, it's very comfortable and a nice design!

**Comment by Tysonpower** [comment]
I had a look at the Sensors in the current code and it seems that we only need three headers?

I would add the following connections to my next Revision, please give me some Feedback if this is enough for the Sensors.

- ONEWIRE_1 for the DS18X20 -> GND SIGNAL 3V3
- two Expansion ports for BME280 and HCSR04 or any other 4 wire module -> GND SIGNAL1 SIGNAL2 3V3

Due to the S3s GPIO matrix i would like to add two general EXP headers that can be used for I2C or any other communication protocol that uses two signal wires.

**Comment by 14yannick** [comment]
Actually my bord look like this. With a row of pinheader at the top for the BME280 module. Main issue with it is that it's to big to fit in the compartment.
![ESP32-DivKitC Pro PCB 2D](https://github.com/Gifford47/HCPBridgeMqtt/assets/6718914/46763d0b-4652-4bed-852a-9c6c54e6bdbf)
I added a second 4 pin row for the HCSR04 in same style as the above.
if there is some place maybe an additional GPIO and gnd would be nice to be able to connect an Reed sensor two have a double check if the door is correctly closed.
Other question would it not make sense to remove the usb connector and uart chip and just prepare a pinout for the UART connection as it's only needed for the initial setup.

**Reply by Tysonpower** [reply]
Okay so you only need the exp ports i mentioned, the reed sensor only needs three pins correct? So GND Signal 3V3?

The ESP32S3 has a integrated USB Controller, so adding a port is really simple and doesen't need much space.

With the additional Ports i should be able to still fit into a 55x55mm pcb, that fits great into my Promatic 4 motor, does the Supramatic motor also has enough room for that?

PS: Placing the BME280 there under the ESP will chnage the Temp values because the ESP produces heat and is very close to the bme.

**Reply by 14yannick** [reply]
The reed just need gpio and ground

**Reply by Tysonpower** [reply]
okay, will just add another 3pin header then, so someone can use a DS18X as well as a reed switch, some reed switch boards have 3 pins i found online so better have a 3V3 extra :)

**Comment by Tysonpower** [comment]
Just ordered new my next PCB batch, whish me luck they work :)

EXP_1: 3V3 GND DATA_1 DATA_2 -> Perfect for the BME280 using a common pin layout for that.
EXP_2: GND DATA_1 DATA_2 3V3 -> For any Sensor with two Data pins
OneWIRE_1 and 2: GND DATA 3V3 -> FOR DS18X or other Sensors that need only one Data line
IO13_to_16: for any additional Thing someone might want's to connect

Also i added four LEDs that are connected to the ESP, maybe we can indicate wifi connection status or other stuff with them in the Future.

The outline is still only 55x55mm, so it should fit great into the motor case

![image](https://github.com/Gifford47/HCPBridgeMqtt/assets/13482963/69488265-b519-4be2-a3a1-92106b98926f)

**Reply by Gifford47** [reply]
cool, that´s great!
my board is intended for external use outside the motor enclosure. this allows the sensors to measure the environment directly.
then we have a corresponding board for both applications :)

**Reply by 14yannick** [reply]
Nice I have ordered some parts to make some experience like this 
https://docs.m5stack.com/en/stamp/stamp_pwr485 or https://www.lilygo.cc/products/t-rsc3.
would you like to share your part list as I'm will order some piece from mouser.

**Reply by Gifford47** [reply]
I don't know if you meant me, but I've posted my bom.
[BOM_PCB_HCPBridge_2023-07-28.csv](https://github.com/Gifford47/HCPBridgeMqtt/files/12193051/BOM_PCB_HCPBridge_2023-07-28.csv)
![pcb](https://github.com/Gifford47/HCPBridgeMqtt/assets/49484063/efac27dc-8991-4626-97f0-a8e4693db43a)

**Reply by 14yannick** [reply]
I'm interrested in all kind of bom as I'm passing an order at mouser and I need 20 EUR more to get the free delivery. Your are using really small resistors and caps (0402). I ordered a batch 0805 to get a bit comfortable with solder SMCs. @Tysonpower where do you do your Step down conversion to 3.3v?

**Reply by Gifford47** [reply]
yes you are right. But all smd components are already assembled. only the esp has to be soldered.

**Reply by Tysonpower** [reply]
on my new pcb it's s small dc dc buck from TI, on the pcb marked U6. But my last pcb had a LM2575S-3.3 similar to the orginal design and the one from gifford

**Comment by Eagle3386** [comment]
Since I just discovered that I can reach the three of you (@14yannick, @Gifford47 & @Tysonpower) right here & all at once, I'd like to summarize my various replies throughout GitHub in the last couple of minutes in 1 comment:

Please, can you guys create a PoE-only layout or at least enable me to use such?
I'd probably don't really care if all that still fits the compartment of a E3 or E4 motor, as long as its PoE-only - I _need_ HA-based control over my garage door (still a ProMatic 1 right now, will be replaced with a SupraMatic E3 or 4, depending on your answers), but also non-wireless communication with the bridge.

Again, I'm begging you guys, please make it happen! Xmas is already approaching! 😉

**Comment by Gifford47** [comment]
Hi!
For my part, I can't afford the time. PoE or Ethernet is not a small feature. It would be a completely new layout AND a completely new firmware. In addition, the esp32 in its current form could not do everything in parallel as we only have two cores and one is already busy communicating with the motor. Timing is crucial here. The same applies to Ethernet. Therefore, I don't see a solution here in terms of performance, time and hardware, sry. Customize your environment or requirements is the only idea I have at the moment.
